python/ray/train/v2/api/train_fn_utils.py

```diff
@@ -3,34 +3,86 @@
 from ray.train import Checkpoint
 from ray.train._internal import session
 from ray.train.context import get_context as get_context_v1
-from ray.train.v2._internal.execution.context import TrainContext, get_train_context
+from ray.train.v2._internal.execution.context import (
+    get_train_context as internal_get_train_context,
+)
 from ray.train.v2._internal.util import _copy_doc
 
 if TYPE_CHECKING:
     from ray.data import DataIterator
 
 
+class TrainContext:
+    @_copy_doc(session.get_metadata)
+    def get_metadata(self) -> Dict[str, Any]:
+        return internal_get_train_context().get_metadata()
+
+    @_copy_doc(session.get_trial_name)
+    def get_trial_name(self) -> str:
+        return internal_get_train_context().get_trial_name()
+
+    @_copy_doc(session.get_trial_id)
+    def get_trial_id(self) -> str:
+        return internal_get_train_context().get_trial_id()
+
+    @_copy_doc(session.get_trial_resources)
+    def get_trial_resources(self):
+        return internal_get_train_context().get_trial_resources()
+
+    @_copy_doc(session.get_trial_dir)
+    def get_trial_dir(self) -> str:
+        return internal_get_train_context().get_trial_dir()
+
+    @_copy_doc(session.get_experiment_name)
+    def get_experiment_name(self) -> str:
+        return internal_get_train_context().get_experiment_name()
+
+    @_copy_doc(session.get_world_size)
+    def get_world_size(self) -> int:
+        return internal_get_train_context().get_world_size()
```

Comment:
can you replace the copy doc with copying the actual docstring for all of the non-deprecated messages?

copy it from the session.get_world_size docstring. This is because code editors do not show docstrings that are dynamically added (e.g. with the copy doc decorator)